src/software/jetson_nano/services/network.cpp

```diff
@@ -34,10 +37,28 @@ void NetworkService::primitiveSetCallback(TbotsProto::PrimitiveSet input)
 {
     std::scoped_lock<std::mutex> lock(primitive_set_mutex);
     primitive_set_msg = input;
+
+    if (last_primitive_sequence_number != input.sequence_number() - 1)
+    {
+        total_primitives_lost++;
```

Comment:
Should this not be the following so we count all the primitives lost (Assuming that there could be more than between two callbacks): 
```cpp
total_primitives_lost += input.sequence_number() - last_primitive_sequence_number - 1
```
If so, we can remove the if statement as well.

Reply:
I believe that implementation will over count the number of primitives lost when multiple primitives are lost sequentially. Assuming the callback is always eventually called the case you mentioned would be handled correctly with the current implementation

Reply:
Can we assume that the callback is always called though? If one primitive can be lost, is it not possible for multiple primitives to be lost back to back?

Reply:
re reading the code your implementation wont over count and is fine for the assumption of callback being dropped